### PR TITLE
feat: automations and tests

### PR DESCRIPTION
**Pull request by JarbasAl**
## Summary by CodeRabbit

- **New Features**
	- Introduced automated workflows for running formatter and unit tests in GitHub Actions.
	- Added comprehensive unit tests for date and time formatting functions across multiple languages, enhancing localization support.
	- Implemented extraction tests for durations and datetime values in various languages, ensuring accurate parsing.

- **Bug Fixes**
	- Improved handling of edge cases and ambiguous time expressions in datetime extraction functions for several languages.

- **Documentation**
	- Updated test files to include detailed comments and structure for better understanding and future enhancements.

### CODE REVIEW

File: test/parse_tests/test_parse_fa.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+        testExtract("هوای امروز بعد از ظهر چطوره؟",
+                    "2017-06-27 15:00:00", "هوای چطوره؟")
+        testExtract("یادم بنداز که هشت هفته و دو روز دیگه به مادرم زنگ بزنم",
+                    "2017-08-24 00:00:00", "یادم بنداز که به مادرم زنگ بزنم")
+        # testExtract("یادم بنداز که دوازده مرداد به مادرم زنگ بزنم",
+        #            "2017-08-03 00:00:00", "یادم بنداز که به مادرم زنگ بزنم")
+        # testExtract("یادم بنداز که ساعت هفت به مادرم زنگ بزنم",
+        #            "2017-06-28 07:00:00", "یادم بنداز که به مادرم زنگ بزنم")
+        # testExtract("یادم بنداز که فردا ساعت بیست و دو به مادرم زنگ بزنم",
+        #            "2017-06-28 22:00:00", "یادم بنداز که به مادرم زنگ بزنم")
```
[the remark this answers is not in the document]
Comment (JarbasAl) [reply test/parse_tests/test_parse_fa.py:109]: open an issue to track this

File: test/format_tests/test_format_it.py [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+        self.assertEqual(nice_time(dt, lang="it", speech=False,
+                                   use_ampm=True), "1:02 PM")
+        self.assertEqual(nice_time(dt, lang="it", speech=False,
+                                   use_24hour=True), "13:02")
+        self.assertEqual(nice_time(dt, lang="it", speech=False,
+                                   use_24hour=True, use_ampm=True), "13:02")
+        self.assertEqual(nice_time(dt, lang="it", use_24hour=True,
+                                   use_ampm=True), "tredici e zero due")
+        self.assertEqual(nice_time(dt, lang="it", use_24hour=True,
+                                   use_ampm=False), "tredici e zero due")
```
[the remark this answers is not in the document]
Comment (JarbasAl) [reply test/format_tests/test_format_it.py:101]: open an issue to track this

File: test/format_tests/test_format_sv.py [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+        dt = datetime.datetime(2017, 1, 31, 4, 50, 00, tzinfo=default_timezone())
+        self.assertEqual(nice_time(dt, lang="sv-se"), "tio i fem")
+
+        dt = datetime.datetime(2017, 1, 31, 5, 55, 00, tzinfo=default_timezone())
+        self.assertEqual(nice_time(dt, lang="sv-se"), "fem i sex")
+
+        dt = datetime.datetime(2017, 1, 31, 5, 30, 00, tzinfo=default_timezone())
+        self.assertEqual(nice_time(dt, lang="sv-se", use_ampm=True),
+                         "halv sex på morgonen")
+
```
[the remark this answers is not in the document]
Comment (JarbasAl) [reply test/format_tests/test_format_sv.py:182]: open an issue to track this

File: test/format_tests/test_format_eu.py [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+
+        dt = datetime.datetime(2017, 1, 31,
+                               23, 15, 9)
+        self.assertEqual(nice_time(dt, lang="eu-eu", use_24hour=True,
+                                   use_ampm=True),
+                         "hogeita hiruak hamabost")
+        self.assertEqual(nice_time(dt, lang="eu-eu", use_24hour=False,
+                                   use_ampm=True),
+                         "gaueko hamaikak eta laurden")
+
```
[the remark this answers is not in the document]
Comment (JarbasAl) [reply test/format_tests/test_format_eu.py:193]: open an issue to track this

File: test/format_tests/test_format_nl.py [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+        dt = datetime.datetime(2017, 1, 31,
+                               5, 55, 00, tzinfo=default_timezone())
+        self.assertEqual(nice_time(dt, lang="nl-nl"),
+                         "vijf voor zes")
+
+        dt = datetime.datetime(2017, 1, 31,
+                               5, 30, 00, tzinfo=default_timezone())
+        self.assertEqual(nice_time(dt, lang="nl-nl", use_ampm=True),
+                         "half zes 's nachts")
+
```
[the remark this answers is not in the document]
Comment (JarbasAl) [reply test/format_tests/test_format_nl.py:202]: open an issue to track this

File: test/format_tests/test_format_hu.py [hunk 6]
```diff
@@ -0,0 +1,205 @@
...
+        dt = datetime.datetime(2017, 1, 31,
+                               5, 55, 00, tzinfo=default_timezone())
+        self.assertEqual(nice_time(dt, lang="hu-hu"),
+                         "öt óra ötvenöt")
+
+        dt = datetime.datetime(2017, 1, 31,
+                               5, 30, 00, tzinfo=default_timezone())
+        self.assertEqual(nice_time(dt, lang="hu-hu", use_ampm=True),
+                         "reggel öt óra harminc")
+
```
[the remark this answers is not in the document]
Comment (JarbasAl) [reply test/format_tests/test_format_hu.py:202]: open an issue to track this

File: test/format_tests/test_format_es.py [hunk 7]
```diff
@@ -0,0 +1,198 @@
...
+
+        dt = datetime.datetime(2017, 1, 31,
+                               23, 15, 9, tzinfo=default_timezone())
+        self.assertEqual(nice_time(dt, lang="es-es", use_24hour=True,
+                                   use_ampm=True),
+                         "las veintitrés quince")
+        self.assertEqual(nice_time(dt, lang="es-es", use_24hour=False,
+                                   use_ampm=True),
+                         "las once y cuarto de la noche")
+
```
[the remark this answers is not in the document]
Comment (JarbasAl) [reply test/format_tests/test_format_es.py:195]: open an issue to track this

File: test/format_tests/test_format_sl.py [hunk 8]
```diff
@@ -0,0 +1,214 @@
...
+        self.assertEqual(nice_duration(500000, "sl"),
+                         "pet dni  osemnajst ur triinpetdeset minut dvajset sekund")  # nopep8
+        self.assertEqual(nice_duration(500000, "sl", speech=False), "5d 18:53:20")
+        self.assertEqual(nice_duration(datetime.timedelta(seconds=500000), "sl", "sl",
+                                       speech=False),
+                         "5d 18:53:20")
+
+
+if __name__ == "__main__":
+    unittest.main()
```
[the remark this answers is not in the document]
Comment (JarbasAl) [reply test/format_tests/test_format_sl.py:214]: open an issue to track this

File: test/format_tests/test_format_ru.py [hunk 9]
```diff
@@ -0,0 +1,268 @@
...
+
+        # Test all years from 0 to 9999 for all languages,
+        # that some output is produced
+        print("Test all years in " + lang)
+        for i in range(1, 9999):
+            dt = datetime.datetime(i, 1, 31, 13, 2, 3,
+                                   tzinfo=default_timezone())
+            self.assertTrue(len(nice_year(dt, lang=lang)) > 0)
+            # Looking through the date sequence can be helpful
+
```
[the remark this answers is not in the document]
Comment (JarbasAl) [reply test/format_tests/test_format_ru.py:242]: open an issue to track this

File: test/format_tests/test_format_uk.py [hunk 10]
```diff
@@ -0,0 +1,269 @@
...
+        self.assertEqual(nice_duration(50000, "uk"),
+                         "тринадцять годин п'ятдесят три хвилини двадцять секунд")
+        #self.assertEqual(nice_duration(50000, "uk", speech=False), "13:53:20")
+        self.assertEqual(nice_duration(500000, "uk"),
+                         "п'ять днів вісімнадцять годин п'ятдесят три хвилини двадцять секунд")  # nopep8
+        #self.assertEqual(nice_duration(500000, "uk", speech=False), "5d 18:53:20")
+        #self.assertEqual(nice_duration(datetime.timedelta(seconds=500000), "uk",
+        #                               speech=False),
+        #                 "5d 18:53:20")
+
```
[the remark this answers is not in the document]
Comment (JarbasAl) [reply test/format_tests/test_format_uk.py:265]: open an issue to track this

File: test/parse_tests/test_parse_az.py [hunk 11]
```diff
@@ -0,0 +1,263 @@
...
+
+        testExtract("5 dəqiqəyə görüşək",
+                    "2017-06-27 10:06:02", "görüşək")
+        testExtract("5 saniyədə görüşək",
+                    "2017-06-27 10:01:07", "görüşək")
+        testExtract("1 saatda görüşək",
+                    "2017-06-27 11:01:02", "görüşək")
+        testExtract("2 saata görüşək",
+                    "2017-06-27 12:01:02", "görüşək")
+
```
[the remark this answers is not in the document]
Comment (JarbasAl) [reply test/parse_tests/test_parse_az.py:260]: open an issue to track this
